Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -1233,8 +1233,17 @@ void SetHighError(u32 high_error)
 void FinishExecutingCommand(ReplyType reply_type, DIInterruptType interrupt_type, s64 cycles_late,
                             const std::vector<u8>& data)
 {
-  s_DIMAR += s_DILENGTH;
-  s_DILENGTH = 0;
+  // The data parameter contains the requested data iff this was called from DVDThread, and is
+  // empty otherwise. DVDThread is the only source of ReplyType::NoReply and ReplyType::DTK.
+
+  u32 transfer_size = 0;
+  if (reply_type == ReplyType::NoReply)
+    transfer_size = static_cast<u32>(data.size());
```

Comment:
Can this value ever exceed the current value of `s_DILENGTH` (making it potentially negative after the subtraction down there)?

Reply:
Only if the emulated software overwrites `s_DILENGTH` in the middle of a read, which doesn't make much sense. I'm not sure what happens on hardware if you do that.

Reply:
Good enough for me then.